gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/Orchestrator.java

```diff
@@ -244,6 +244,7 @@ public void orchestrate(Spec spec, Properties jobProps, long triggerTimestampMil
         return;
       }
       Map<String, String> flowMetadata = TimingEventUtils.getFlowMetadata((FlowSpec) spec);
+      FlowCompilationValidationHelper.addFlowExecutionIdIfAbsent(flowMetadata, jobExecutionPlanDagOptional.get());
```

Comment:
seems an important property to add.

on that note, do we have any existing `Orchestrator.orchestrate` test case we might be able to augment with an assert that this is in fact set?  (e.g. using test inputs that would fail that assert w/o the above code change.)